kratos/includes/mortar_classes.h

```diff
@@ -489,23 +489,26 @@ class MortarKinematicVariablesWithDerivatives : public MortarKinematicVariables<
  * This data will be used to compute the derivatives.
  * This class includes different information that is used in order to compute the directional derivatives in the mortar contact conditions
  */
-template< unsigned int TDim, unsigned int TNumNodes>
+template< unsigned int TDim, unsigned int TNumNodes, bool TNormalVariation>
 class DerivativeData
 {
 public:
     ///@name Type Definitions
     ///@{
     
     // Auxiliar types
-    typedef array_1d<double, TNumNodes>                                         type_1;
+    typedef bounded_matrix<int, 1, 1>                    type_0;
+    typedef array_1d<double, TNumNodes>                  type_1;
     typedef bounded_matrix<double, TNumNodes, TDim>      type_2;
     typedef bounded_matrix<double, TNumNodes, TNumNodes> type_3;
-    typedef bounded_matrix<double, 3, 3>                 type_4;
+    typedef typename std::conditional<TNumNodes == 2, type_0, bounded_matrix<double, 3, 3>>::type type_4;
```

Comment:
what are u doing here?

Reply:
Defining and zero size matrix in the cases is not use, to reduce the amount of memory used